nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -853,11 +852,7 @@ in {
               ],
               ${optionalString (showAppStoreSetting) "'appstoreenabled' => ${renderedAppStoreSetting},"}
               'datadirectory' => '${datadir}/data',
```

Comment:
This could be e.g. `services.nextcloud.extraOptions.datadirectory = lib.mkDefault "${datadir}/data";` then.